opslevel/common.go

```diff
@@ -32,6 +35,36 @@ import (
 
 // var DefaultPredicateDescription = "A condition that should be satisfied."
 
+const (
+	providerIssueUrl = "https://github.com/OpsLevel/terraform-provider-opslevel/issues"
+	providerFile     = "../tests/provider.tf"
+)
+
+type CommonClient struct {
+	client *opslevel.Client
+}
+
+// Configure sets up the OpsLevel client for datasources and resources
+func (d *CommonClient) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
```

Comment:
what are the cases where the provider would be nil?

Reply:
I'm not really sure tbh. This is pretty much copy pasta from the Terraform example 😁